aspnetcore/migration/70-80.md

```diff
@@ -73,64 +73,541 @@ In the project file, update each [Microsoft.AspNetCore.*](https://www.nuget.org/
+
+   ```diff
+   - <script src="_framework/blazor.webassembly.js"></script>
+   + <script src="_framework/blazor.web.js"></script>
+   ```
+
+1. Remove the following line from `Client/Program.cs`:
+
+   ```diff
+   - builder.RootComponents.Add<HeadOutlet>("head::after");
+   ```
+   
+1. Update `Server/Program.cs`:
+
+   Add a `using` statement for the `Host` component at the top of the file.
+
+   > [!NOTE]
+   > In the following example, the project's namespace is `Net8BlazorWebAssembly.Client`. Adjust the namespace to match your project.
+
+   ```csharp
+   using Host = Net8BlazorWebAssembly.Client.Host;
+   ```
+
+   Add Razor component and interactive WebAssembly component services to the project. Call `AddRazorComponents` with a chained call to `AddInteractiveWebAssemblyComponents`:
+
+   ```csharp
+   builder.Services.AddRazorComponents()
+       .AddInteractiveWebAssemblyComponents();
+   ```
+
+   Add [Antiforgery Middleware](xref:blazor/security/index#antiforgery-support) to the request processing pipeline immediately after the call to `app.UseRouting`. If there are calls to `app.UseRouting` and `app.UseEndpoints`, the call to `app.UseAntiforgery` must go between them. Calls to `app.UseAntiforgery` must be placed after calls to `app.UseAuthentication` and `app.UseAuthorization`.
+
+   ```csharp
+   app.UseAntiforgery();
+   ```
+
+   Remove the following line:
+   
+   ```diff
+   - app.UseBlazorFrameworkFiles();
+   ```
+
+   Remove the following line:
+   
+   ```diff
+   - app.MapFallbackToFile("index.html");
+   ```
+     
+   Replace the preceding line with with a call to `MapRazorComponents`, supplying the `Host` component as the root component type, and add a chained call to `AddInteractiveWebAssemblyRenderMode`:
+   
+   ```csharp
+   app.MapRazorComponents<Host>()
+       .AddInteractiveWebAssemblyRenderMode();
+   ```
+
+1. To enable prerendering:
+
+   Update the app's Razor components to support prerendering from the server.
 
-For more information, see the following content in the *Data binding* article:
+   <!-- What's in the sample is highly template-specific (the weather data example) 
+        and won't match what they need to do generally. What do we tell devs
+        for this step? -->
 
-* [Introduction](xref:blazor/components/data-binding?view=aspnetcore-7.0)
-* [Bind across more than two components](xref:blazor/components/data-binding?view=aspnetcore-7.0#bind-across-more-than-two-components)
+   In the `Client/Host.razor` file, remove the following markup:
+   
+   ```diff
+   - <div id="app">
+   -     <svg class="loading-progress">
+   -         <circle r="40%" cx="50%" cy="50%" />
+   -         <circle r="40%" cx="50%" cy="50%" />
+   -     </svg>
+   -     <div class="loading-progress-text"></div>
+   - </div>
+   ```
+     
+   Replace the preceding markup with the following line:
+     
+   ```razor
+   <App @rendermode="RenderMode.WebAssembly" />
+   ```
```

Comment:
Let's not document here how to enable prerendering. It involves a fair amount of complexity because the app's components may not be designed to run from the server. Instead, we should guide users to disable prerendering at least initially when converting Blazor WebAssembly apps.

```suggestion
```